The BFD target elf32-little has no specified machine, which trips the
architecture sanity test.  Use elf32-littlearm to set the machine
correctly.

Upstream-Status: Backport
Signed-off-by: Ross Burton <ross.burton@arm.com>

diff --git a/cmake/Common/CompilerGNUARMCommon.cmake b/cmake/Common/CompilerGNUARMCommon.cmake
index 32e805bb..2d3ea8e4 100644
--- a/cmake/Common/CompilerGNUARMCommon.cmake
+++ b/cmake/Common/CompilerGNUARMCommon.cmake
@@ -196,7 +196,7 @@ function(compiler_generate_hex_output TARGET)
 endfunction()
 
 function(compiler_generate_elf_output TARGET)
-	add_custom_command(TARGET ${TARGET} POST_BUILD COMMAND ${CMAKE_GNUARM_OBJCOPY} ARGS -O elf32-little $<TARGET_FILE:${TARGET}> $<TARGET_FILE_DIR:${TARGET}>/${TARGET}.elf)
+	add_custom_command(TARGET ${TARGET} POST_BUILD COMMAND ${CMAKE_GNUARM_OBJCOPY} ARGS -O elf32-littlearm $<TARGET_FILE:${TARGET}> $<TARGET_FILE_DIR:${TARGET}>/${TARGET}.elf)
 endfunction()
 
 # Function for creating a new target that preprocesses a .c file
